tests/vmi_multus_test.go

```diff
@@ -1008,16 +1046,56 @@ var _ = Describe("[Serial]Macvtap", func() {
 
 		BeforeEach(func() {
 			macAddress := "02:03:04:05:06:07"
-			macvtapVMI = createCirrosVMIWithMacvtapDefinedMAC(virtClient, macvtapNetworkName, macAddress)
+			clientVMI = createCirrosVMIStaticIPRandomNode(virtClient, macvtapNetworkName, macAddress)
 		})
 
 		It("should be successful when the VMI MAC address is defined in its spec", func() {
 			By("starting the migration")
-			migration := tests.NewRandomMigration(macvtapVMI.GetName(), macvtapVMI.GetNamespace())
+			migration := tests.NewRandomMigration(clientVMI.GetName(), clientVMI.GetNamespace())
 			migrationUID := tests.RunMigrationAndExpectCompletion(virtClient, migration, migrationWaitTime)
 
 			// check VMI, confirm migration state
-			tests.ConfirmVMIPostMigration(virtClient, macvtapVMI, migrationUID)
+			tests.ConfirmVMIPostMigration(virtClient, clientVMI, migrationUID)
+		})
+
+		Context("with live traffic", func() {
+			var serverVMI *v1.VirtualMachineInstance
+			var serverIP string
+
+			getVMIPSecondaryIface := func(vmName string, namespace string) *string {
+				serverVMI, err = virtClient.VirtualMachineInstance(namespace).Get(vmName, &v13.GetOptions{})
+				if err != nil {
+					return nil
+				}
+				serverIP := serverVMI.Status.Interfaces[secondaryIfaceIndex].IP
+				if serverIP != "" {
+					return &serverIP
+				}
+				return nil
+			}
+
+			BeforeEach(func() {
+				macAddress := "02:03:04:05:06:aa"
+
+				serverVMI = createFedoraVMIRandomNode(virtClient, macvtapNetworkName, macAddress)
+				Expect(serverVMI.Status.Interfaces).NotTo(BeEmpty(), "a migrate-able VMI must have network interfaces")
+				Eventually(getVMIPSecondaryIface, 2*time.Minute, time.Second).ShouldNot(BeNil(), "a migrate-able VMI must have network interfaces")
+
+				ipAndNetworkLength := strings.Split(serverVMI.Status.Interfaces[secondaryIfaceIndex].IP, "/")
```

Comment:
You can use the helper function here, no?

If you do, primary/secondary constants can be dropped.

Reply:
I've re-worked the `createRunningVMIWithMacOnPrimaryAddr` so it returns the IP address.

Also removed the eventually, and instead block on the helper, which allows to remove the side effect of updating the `serverVMI` in the helper and removes the need to re-read the VMI entity once we know the IP is ready.

This is a separate commit so I can easily roll-back if this approach is not desired. Let me know your thoughts.